### PR TITLE
fix: set cmp preselect to none.

### PR DESCRIPTION
**Pull request by ayamir**
#1246

### CODE REVIEW

**Comment by JohnWickzzz** [comment]
image

After update preselect like this and add completeopt param. Simple test of java/go/lua file, now it can select the first item. But I don't know if it can run normally in all places.

File: lua/modules/configs/completion/cmp.lua [hunk 1]
```diff
@@ -66,7 +66,7 @@ return function()
 
 	local cmp = require("cmp")
 	require("modules.utils").load_plugin("cmp", {
-		preselect = cmp.PreselectMode.Item,
+		preselect = cmp.PreselectMode.None,
```
Comment (Jint-lzxy) [inline lua/modules/configs/completion/cmp.lua:69]: qq: I think given the current changes, we could introduce a new keymap to commit the default selection, similar to the following using `<C-CR>` to replicate the original behavior:
```lua
["<C-CR>"] = cmp.mapping.confirm({ select = true, behavior = cmp.ConfirmBehavior.Insert }),
```

Comment (ayamir) [reply lua/modules/configs/completion/cmp.lua:69]: This PR mainly fixed the cmp preselect item inconsistency when editing different lang codes. Now preselect is set to NONE. IMO it has nothing to do with a new keymap to commit the default preselect. Additionly, seems my terminal can't distinguish `C-CR` from `CR`.

Comment (Jint-lzxy) [reply lua/modules/configs/completion/cmp.lua:69]: Agreed.

**Pull request merged**